src/lm/kaldi-lmtable.cc

```diff
@@ -207,15 +208,24 @@ bool LmTable::ReadFstFromLmFile(std::istream &istrm,
     }
     // found valid line
     ngram_order = atoi(inpline.substr(pos1+5, pos2-(pos1+5)).c_str());
+    orders.push_back(ngram_order);
     if (ngram_order > max_ngram_order) {
       max_ngram_order = ngram_order;
     }
   }
+  std::sort(orders.begin(), orders.end());
```

Comment:
Also,  just for consideration : some tools report the number  of ngrams
slightly different  than the real number of ngrams in the appropriate
section.
Something  like that can happen even in kaldi,  for example in arpa2G.sh
we  remove some presumably faulty/invalid ngrams only using grep -v.
So perhaps just issue warning instead of error,  if the real count and the
number from the header differ just by little?
I don't have any strong opinions.
Y.
On Nov 25, 2015 10:58 PM, "Daniel Povey" <notifications@github.com> wrote:

> In src/lm/kaldi-lmtable.cc
> <https://github.com/kaldi-asr/kaldi/pull/365#discussion_r45942604>:
>
> >      if (ngram_order > max_ngram_order) {
> >        max_ngram_order = ngram_order;
> >      }
> >    }
> > +  std::sort(orders.begin(), orders.end());
>
> this should not be necessary as the 1-grams should precede the 2-grams
> (etc.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/365/files#r45942604>.
>